src/containers/BooksForm.js

```diff
@@ -8,21 +13,48 @@ const BooksForm = () => {
     'Learning',
     'Sci-Fi',
   ];
+  const [state, setState] = useState({ title: '', category: 'Action' });
+  let error = false;
+
+  const handleChange = ({ target: { name, value } }) => {
+    setState({ ...state, [name]: value });
+  };
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const { title, category } = state;
+    if (title === '' || category === '') {
+      error = true;
+    } else {
+      createBook({
+        id: Math.round(Math.random() * 100) + 1,
+        title,
+        category,
+      });
+      setState({ title: '', category: 'Action' });
+    }
+    return error;
+  };
 
   return (
     <>
-      <form className="form">
+      <form className="form" onSubmit={handleSubmit}>
         <h3>Add book</h3>
         <div className="title-div">
           <label htmlFor="book-title">
             Title:
-            <input type="text" id="book-title" />
+            <input
+              type="text"
+              id="book-title"
+              name="title"
+              onChange={handleChange}
+              value={state.title || ''}
+            />
           </label>
         </div>
         <div className="category-div">
           <label htmlFor="book-category">
             Category:
-            <select id="book-category">
+            <select id="book-category" onChange={handleChange} name="category">
```

Comment:
- [ ] [REQUIRED] To make sure that your APP works without bug, please change this line to: 
```JSx
<select id="book-category" value={state.category} onChange={handleChange} name="category">
```
The value of the select option will change after each update even if, the onChange event is not called.

Reply:
Done